tests/main/failover/task.yaml

```diff
@@ -43,6 +44,36 @@ execute: |
         truncate -s 0 /tmp/unpack/lib/systemd/systemd
     }
 
+    inject_systemdloop_failure(){
+        cat <<EOF > /tmp/unpack/lib/systemd/system/deadlock.service
+    [Unit]
+    Before=sysinit.target
+    DefaultDependencies=no
+    [Service]
+    Type=oneshot
+    ExecStartPre=-/bin/sh -c "echo 'DEBUG: $(date): deadlocked system' >/dev/console"
+    ExecStartPre=-/bin/sh -c "echo 'DEBUG: $(date): deadlocked system' >/dev/ttyS0"
+    ExecStart=/bin/systemctl start deadlock.service
+    RemainAfterExit=yes
+    [Install]
+    RequiredBy=sysinit.target
+    EOF
+        cat <<EOF > /tmp/unpack/lib/systemd/system/autoreboot.service
+    [Unit]
+    DefaultDependencies=no
+    Description=Force reboot if booting did not finish after 20s (the system does not do it by itself)
```

Comment:
The test is nice but it also highlights a gap in our failover strategy. The needs for this auto-reboot unit in the test is something that ideally snapd/snappy should handle itself. But we all know this is tricky. Not sure what to do with the test in the meantime. We could merge it as its a useful start but a note somehwere that this the story is not complete may be a good idea.